tokio/src/task/harness.rs

```diff
@@ -151,8 +150,12 @@ where
     pub(super) unsafe fn drop_task(mut self) {
         let might_drop_join_waker_on_release = self.might_drop_join_waker_on_release();
 
-        // Read the join waker cell just to have it
-        let (join_waker, check) = self.read_join_waker();
```

Comment:
Removing `check` removes assertions during the `loom` tests. Is there any way to keep `check` around?

Reply:
The `read_join_waker` implementation now uses `CausalCell::with` rather than `CausalCell::with_deferred` so the check is still performed.